projects/dotcms-field-elements/src/components/dot-binary-file/dot-binary-file.tsx

```diff
@@ -0,0 +1,312 @@
+                this.el.classList.remove('dot-dragover');
+                const droppedFile: File = evt.dataTransfer.files[0];
+                if (this.isFileAllowed(droppedFile.name)) {
+                    this.setValue(droppedFile);
+                    this.setFileDisplayName(droppedFile.name);
+                } else {
+                    this.clearField(this.validationMessage);
+                }
+            }
+        });
+    }
+
+    // only supported in iOS.
+    private pasteHandler(): void {
+        this.el.addEventListener('paste', (evt: ClipboardEvent) => {
+            const clipboardData: DataTransfer = evt.clipboardData;
+            if (clipboardData.items.length) {
+                if (this.isPastingFile(clipboardData)) {
+                    this.handleFilePaste(clipboardData.items);
+                } else {
+                    this.handleURLPaste(clipboardData.items[0]);
+                }
+            }
+        });
+    }
+
+    private changeHandler(event: Event): void {
+        const input: HTMLInputElement = event.srcElement as HTMLInputElement;
+        if (this.isFileAllowed(input.files[0].name)) {
+            this.setValue(input.files[0]);
+            this.setFileDisplayName(input.files[0].name);
+        } else {
+            event.preventDefault();
+            this.clearField(this.validationMessage);
+        }
+    }
+
+    private handleFilePaste(items: DataTransferItemList) {
+        const clipBoardFile = items[1].getAsFile();
+        items[0].getAsString((fileName: string) => {
+            if (this.isFileAllowed(fileName)) {
+                this.setValue(clipBoardFile);
+            } else {
+                this.clearField(this.validationMessage);
+            }
+        });
+    }
+
+    private handleURLPaste(item: DataTransferItem) {
+        item.getAsString((fileURL: string) => {
+            if (this.isValidURL(fileURL)) {
+                this.setValue(fileURL);
+            } else {
+                this.clearField(this.URLValidationMessage);
+            }
+        });
+    }
+
+    private clearField(validationMessage: string) {
+        this.setValue(null, validationMessage);
+        this.setFileDisplayName('');
+    }
+
+    private isValidURL(url: string): boolean {
+        return URL_REGEX.test(url);
+    }
+
+    private isPastingFile(data: DataTransfer): boolean {
+        return !!data.files.length;
+    }
+
+    private keyDownHandler(evt: KeyboardEvent): void {
+        if (evt.key === 'Backspace') {
+            this.required ? this.setValue(null, this.requiredMessage) : this.setValue(null);
+        }
+    }
+
+    private buttonHandler(): void {
+        const fileInput: HTMLInputElement = this.el.querySelector('dot-label input[type="file"]');
+        fileInput.click();
+    }
+
+    private setFileDisplayName(name: string): void {
+        const textInput: HTMLInputElement = this.el.querySelector('dot-label input[type="text"]');
+        textInput.value = name;
```

Comment:
Why this is not a property that we can pass in the html of the render? why we need to do it manually?